core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -208,6 +209,10 @@ class TransactionCoordinator(brokerId: Int,
     if (transactionalId == null || transactionalId.isEmpty) {
       responseCallback(Errors.INVALID_REQUEST)
     } else {
+      // if there is any partitions unknown in the metadata cache, return immediately to client
+      if (partitions.exists(tp => !metadataCache.contains(tp.topic, tp.partition)))
+        responseCallback(Errors.UNKNOWN_TOPIC_OR_PARTITION)
```

Comment:
You are right. In other words, the problem is that the `UNKNOWN_TOPIC_PARTITION` is a retriable error right now, so it will keep retrying. We need to add logic distinguish between cases where a broker is bounced and doesn't have its metadata yet, and from cases where the topic is truly deleted. There is no such logic in the client today, so it will retry indefinitely when a topic is deleted.